### DISCUSSION TITLE
how to use google authentication using laravel 8 backend

### DISCUSSION THREAD

**Discussion by rajeshtva** [quasarframework/quasar#10546] [General - Components / Directives / etc]
I am working on an **web applicaiton** (spa) where we need to authenticate user using google , facebook OAuth2 authentication system. till now i couldn't find any tutorials on it. i am using Quasar v2.0.3. with **laravel 8** backend 
thanks for any help

**Comment by ontwikkelfabriek** [comment]
@je-rajesh 

Maybe this is better suited for your purpose ( laravel 8 + vue integrated == easy oauth):
https://inertiajs.com/

You could use Quasar as a Vue plugin or in UMD mode.

**Comment by MatthewSH** [comment]
Have you looked at [Sanctum](https://laravel.com/docs/8.x/sanctum#introduction)? Unless you don't store any user information in the database...you'll probably have a user model. Sanctum is built for things like SPAs.

I'll be honest, I'm not quite sure why @ontwikkelfabriek is suggesting Intertia as that's just another JS framework. It wouldn't really help because you still have to create the same things, you may have an easier time doing SSR stuff because of their packages, but past that you'll still be doing a lot of groundwork for your specific use-case. Unless I'm missing something and if I'm missing something crucial about Interia that really does suit it better for the job then please feel free to correct me. 
I've done mixed Vue/Blade applications with Laravel before they switched to semver (so this is Laravel 5.x era) and I'd easily go back to it if I had an application that fit the requirements.

~As for recommending using Intertia then using Quasar in UMD or a Vue plugin...sure you could but that's adding complexity that could prove to be a major pain point long term. I'd recommend against it in this specific instance (if you go forward with Intertia) because you'll potentially be hurting yourself in the future.~
EDIT: Although the top could in theory apply, Inertia looks like it targets implementation a bit differently and that was my misunderstanding. You're honestly better off, IMO, developing a front end and then developing a backend separately...how you integrate those 2 is up to you but either way...I really don't see the reason you'd need Inertia at all. It looks like it could just collide more with the way Quasar does it's routing but I could be wrong. Inertia just seems like, at its core, a routing library which to me makes less sense why it's being suggested because it doesn't really answer your question about the implementation of the OAuth authorization with a Quasar/Vue front end. 🤷‍♂️ Unless, again, I'm just missing something.

I'd say, go with Sanctum and Socialite first; if that doesn't work...then move on.
A simple google search of: 
> laravel sanctum with socialite

Returned [this article](https://bipinmhz.medium.com/sanctum-with-socialite-api-authentication-via-social-networks-in-laravel-8-624cb1737e8) which should help guide you in the general direction. As with everything, take it with a grain of salt and always consult the official docs. Taylor and his team do a great job with the docs IMO and they're definitely more helpful than tutorials or Google searches at times.

**Comment by metalsadman** [comment]
look into these repos, author implemented sanctum into quasar  https://github.com/training-yoyosan/example-frontend
https://github.com/training-yoyosan/example-backend.
